handler/OpenIDHandler.php

```diff
@@ -65,7 +67,7 @@ public function __construct(protected OpenIDPlugin $plugin)
 	 */
 	function doAuthentication($args, $request, $provider = null)
 	{
-		$selectedProvider = $request->getUserVar('provider');
+		$selectedProvider = OpenIDProvider::tryFrom($request->getUserVar('provider'));
```

Comment:
This is a very tidy way to validate user input to a vocabulary using enums! What do you think about just using `from` instead of `tryFrom`? That way it'll throw a `ValueError`, which is an appropriate exception to raise. Then there's no need to even consider what to do if `tryFrom` returns `null`.